### PR TITLE
perf: Reduce memory consumption

### PR DESCRIPTION
**Pull request by CarloVarni**
We have three vectors (of `std::size_t`) that call a reserve of these values. Reduce the number to 100 since it makes no sense we have 1000000000 seed candidates for a single middle SP ... not even for ITk at PU200

/cc @LuisFelipeCoelho

### CODE REVIEW

**Review by LuisFelipeCoelho — APPROVED** [review]

**Comment by CarloVarni** [comment]
@LuisFelipeCoelho you have to approve again since there have been commit after the last approval

**Review by LuisFelipeCoelho — APPROVED** [review]

**Pull request merged**
